src/corehost/cli/test/mockcoreclr/mockcoreclr.cpp

```diff
@@ -35,7 +35,7 @@ SHARED_API pal::hresult_t STDMETHODCALLTYPE coreclr_initialize(
 
     if (hostHandle != nullptr)
     {
-        *hostHandle = (coreclr_t::host_handle_t*)(size_t) 0xdeadbeef;
+        *hostHandle = reinterpret_cast<coreclr_t::host_handle_t*>(0xdeadbeef);
```

Comment:
I am a little confused about this piece of code. I would have expected it to be

    hostHandle = reinterpret_cast<coreclr_t::host_handle_t*>(0xdeadbeef);

The initial `*` in the line.. does that mean we write `0xdeadbeef` to the location pointed to by `hostHandle`? I would have expected that we set the `hostHandle` pointer itself to `0xdeadbeef`.

Reply:
The `hostHandle` is an out parameter effectively. It's a pointer to a handle (and the function returns the handle).
I think the cast here is actually wrong (before and after). We're assigning the specific value to the location `hostHandle` points to. Type-wise, it points to `coreclr_t::host_handle_t` (no pointer).
I think it works because that is really just a typedef to `void *`.

Can you please change this to:
```C++
*hostHandle = reinterpret_cast<coreclr_t::host_handle_t>(0xdeadbeef);
```

Reply:
Ah, right. That makes sense.